webui/src/lib/components/repository/TableDiff.jsx

```diff
@@ -0,0 +1,123 @@
+    return <>
+        {(otfDiffs.length === 0) ?  <Alert variant="info" style={{margin: 0+"px"}}>No changes</Alert> :
+            <Table className="table-diff" size="md">
+                <tbody>
+                {
+                    response.results.map(otfDiff => {
+                        return <OtfDiffRow key={otfDiff.timestamp + "-diff-row"} otfDiff={otfDiff}/>;
+                    })
+                }
+                </tbody>
+            </Table>
+        }
+    </>
+}
+
+const OtfDiffRow = ({otfDiff}) => {
+    const [rowExpanded, setRowExpanded] = useState(false);
+    const rowClass = "otf-diff-" + operationToDiffType(otfDiff.operation);
+    return <>
+        <OperationMetadataRow className={rowClass} otfDiff={otfDiff} operationExpanded={rowExpanded} onExpand={() => setRowExpanded(!rowExpanded)}/>
+        {rowExpanded
+            ? <OperationDetailsRow otfDiff={otfDiff}/>
+            : ""}
+    </>
+}
+
+const OperationMetadataRow = ({otfDiff, operationExpanded, onExpand, ...rest}) => {
+    return <tr {...rest}>
+        <td className={"table-operation-type pl-lg-10 col-10"}>{otfDiff.operation}</td>
+        <td className="table-version col-sm-auto">Version = {otfDiff.version}</td>
+        <td className="operation-expansion col-sm-auto">
+            <OperationExpansionSection operationExpanded={operationExpanded} onExpand={onExpand}/>
+        </td>
+    </tr>
+
+}
+
+const OperationDetailsRow = ({otfDiff}) => {
+    const operationTimestamp = otfDiff.timestamp;
+    const operationContent = parseOperationContent(otfDiff.operation_content);
+    return <tr className="otf-diff-operation-details">
+        <td className="pl-lg-10 col-10 table-operation-details">
+            <strong>Timestamp:</strong> {operationTimestamp}
+            <br/>
+            <strong>Commit Info:</strong>
+            <br/>
+            <pre>{operationContent}</pre>
+        </td>
+    </tr>
+}
+
+const OperationExpansionSection = ({operationExpanded, onExpand}) => {
+    return <OverlayTrigger placement="bottom" overlay={<Tooltip>{operationExpanded ? "Hide operation info" : "Show operation info"}</Tooltip>}>
+        <Button variant="link" className="table-operation-expansion" onClick={onExpand}>{operationExpanded ? <ChevronDownIcon/> : <ChevronRightIcon/>}</Button>
+    </OverlayTrigger>
+}
+
+function operationToDiffType(operation) {
+    const diffType = deltaLakeOperationToDiffType.get(operation);
+    return diffType !== undefined ? diffType : DiffType.Changed;
+}
+
+/**
+ * Parses otf operation content. operations sometimes include unparsed json strings and this methods aims to simplify
+ * reading those operations.
+ *
+ * @param content of the table operation
+ * @return a prettified string including the operation content.
+ */
+function parseOperationContent(content) {
+    let parsedContent = "";
+    const JSONContentStr = JSON.stringify(content, null, 2);
+    const JSONContent = JSON.parse(JSONContentStr);
```

Comment:
You are right, this is not needed, because I'm parsing here the following object included in the otfDiff response:
```
operation_content:
          type: object
          description: free form content describing the returned operation diff
```
and the object is a map of key-value pairs, so I don't need to convert it into a json first. 
@Jonathan-Rosenberg should we clarify it in API doc? in it not relly a free form but a collection of key-value pairs.